cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java

```diff
@@ -85,7 +89,7 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
       }
 
       try {
-        taskRunner.run(Id.Program.from(namespaceId, applicationId, programType, programId),
+        taskRunner.run(new ApplicationId(namespaceId, applicationId, appVersion).program(programType, programId),
```

Comment:
why is there no ``ProgramId(String namespace, String app, String version, String type, String program)``?